src/datachain/catalog/catalog.py

```diff
@@ -988,6 +988,14 @@ def create_new_dataset_version(
         schema = {
             c.name: c.type.to_dict() for c in columns if isinstance(c.type, SQLType)
         }
+
+        job_id = job_id or os.getenv("DATACHAIN_JOB_ID")
+        if not job_id:
+            from datachain.query.session import Session
+
+            session = Session.get(catalog=self)
```

Comment:
it seems a bit strange that we are initializing Session here? why is it needed ... can we not do this (we are spreading logic for Session logic management into lower lever functions which is not good)

here for example, reading this is not enough - there are some assumptions about job_id and or Session that are not clear and it's not clear why this code exits at all tbh

Reply:
Session.get gives us the global session if it already exists without initializing.

Reply:
why do we do this only if job_id is not initialized?

what about non-global sessions (query might belong to it)?

Reply:
If we already have job id, it is the scenario of Studio. In such case, we will have the job id to identify the dataset versions created during this job. 

I couldn't find a way to find the dataset versions created by a session. So, I used the existing job id field to identify the versions created during current run. We dont have a way to get current local session context in meta store.

Reply:
okay, so we have 3 cases:

- job_id explicitly provided
- env var
- one from the global Session

do we need all three?
can we make it always specified explicitly?
what if session is local / created by a user? will it still work?

my point here is that this code looks too complicated, how can we isolate if from here? move it to the session itself, etc, etc? are there any reasonable ways? :)

Reply:
> my point here is that this code looks too complicated, how can we isolate if from here? move it to the session itself, etc, etc? are there any reasonable ways? :)

Here, we are just fetching the session and using its job id. IMO, it is not that complicated or there is anything we can move to the session. We are attaching the job id if we don't have it passed with us either by argument or by env variable.